tests/test_flow_runners.py

```diff
@@ -1514,6 +1514,7 @@ async def test_creates_job(
                                 ],
                             }
                         ],
+                        "serviceAccountName": "default",
```

Comment:
With the change to make this an optional setting, the resulting job spec should be unchanged (no `serviceAccountName` by default). This helps us avoid running afoul of admission controller hooks that may want to enforce particular service accounts or use of namespace-default service accounts for whatever reason.